assets/js/components/notifications/FirstPartyModeSetupBanner.js

```diff
@@ -67,21 +67,22 @@ export default function FirstPartyModeSetupBanner( { id, Notification } ) {
 		select( CORE_NOTIFICATIONS ).isNotificationDismissed( id )
 	);
 
-	const { dismissNotification, invalidateResolution } =
-		useDispatch( CORE_NOTIFICATIONS );
+	const { invalidateResolution } = useDispatch( CORE_NOTIFICATIONS );
 	const { setValue } = useDispatch( CORE_UI );
 
-	const onCTAClick = () => {
+	const onCTAClick = async () => {
 		setFirstPartyModeEnabled( true );
-		saveFirstPartyModeSettings();
+		const { error } = await saveFirstPartyModeSettings();
+
+		if ( error ) {
+			return { error };
+		}
 
 		setValue( FPM_SHOW_SETUP_SUCCESS_NOTIFICATION, true );
 		invalidateResolution( 'getQueuedNotifications', [
 			viewContext,
 			NOTIFICATION_GROUPS.DEFAULT,
 		] );
-
-		dismissNotification( id );
```

Comment:
This is a cleanup that is unrelated to the error handling.

We shouldn't need to dismiss the notification here as it's already dismissed in the `Dismiss` component:

https://github.com/google/site-kit-wp/blob/faac928ab9aaf2a702c50fa3cdafeda166df8cde/assets/js/googlesitekit/notifications/components/common/Dismiss.js#L49-L59